Neos.ContentRepository.Core/Classes/Projection/ContentGraph/AbsoluteNodePath.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Projection\ContentGraph;
+
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Exception\AbsoluteNodePathIsInvalid;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeName;
+
+/**
+ * An absolute node path, composed of the root node's type and a relative path from there.
+ *
+ * It describes the hierarchy path of a node to and including its root node in a subgraph.
+ * @api
+ */
+final class AbsoluteNodePath implements \JsonSerializable
+{
+    private function __construct(
+        public readonly NodeTypeName $rootNodeTypeName,
+        public readonly NodePath $path
+    ) {
+    }
+
+    public static function fromComponents(
```

Comment:
Not that important but I suggest to make this more explicit even:
```suggestion
    public static function fromRootNodeTypeNameAndPath(
```